src/PublicAPI/index.php

```diff
@@ -159,6 +160,31 @@ function invokeArgsNamed(ReflectionMethod $refmethod, $object, Array $args = [])
 }
 
 /**
+ * Check and provide Access-Control-Allow-Origin if needed
+ */
+if (isset($_SERVER['HTTP_ORIGIN'])) {
+    if (
+        empty(Config::$api_allowed_domains) or
+        in_array($_SERVER['HTTP_ORIGIN'], Config::$api_allowed_domains)
+    ) {
+        header('Access-Control-Allow-Origin: *');
```

Comment:
potentially use the origin that was sent? seems to generally be recommended over * - even tho it is restricted already